cpp/arcticdb/storage/mongo/mongo_storage-inl.hpp

```diff
@@ -85,12 +85,25 @@ inline void MongoStorage::do_remove(Composite<VariantKey>&& ks, RemoveOpts) {
     auto fmt_db = [](auto &&k) { return variant_key_type(k); };
     ARCTICDB_SAMPLE(MongoStorageRemove, 0)
 
-    (fg::from(ks.as_range()) | fg::move | fg::groupBy(fmt_db)).foreach([&](auto &&group) {
+    Composite<VariantKey> failed_deletes;
+
+    (fg::from(ks.as_range()) | fg::move | fg::groupBy(fmt_db)).foreach([this, &failed_deletes](auto &&group) {
         for (auto &k : group.values()) {
             auto collection = collection_name(variant_key_type(k));
-            client_->remove_keyvalue(db_, collection, k);
+            try {
+                client_->remove_keyvalue(db_, collection, k);
+            } catch(KeyNotFoundException& e) {
+                failed_deletes.push_back(std::move(e.keys()));
+            } catch(const std::exception& e) {
+                // TODO https://github.com/man-group/ArcticDB/issues/518
```

Comment:
Are you sure it's safe to catch here and continue deleting the other stuff? You seem worried about the same thing in issue #518, but the original implementation seems to bail out here if any of the deletes fail?

I might be misunderstanding, since your ticket does say,

```
Currently, do_remove_impl in S3 (and MongoStorage::do_remove) swallows all errors in the batching loop and reports the keys in the erroring requests in a KeyNotFoundException at the end.
```